Tools/WebKitTestRunner/wpe/EventSenderProxyClientWPE.cpp

```diff
@@ -303,6 +303,114 @@ void EventSenderProxyClientWPE::keyDown(WKStringRef keyRef, double time, WKEvent
     wpe_event_unref(event);
 }
 
+#if ENABLE(TOUCH_EVENTS)
+void EventSenderProxyClientWPE::addTouchPoint(int x, int y, double)
+{
+    uint32_t id = 0;
+    for (const auto& point : m_touchPoints) {
+        if (point.id == id)
+            id++;
+    }
+    m_touchPoints.append({ id, TouchPoint::State::Down, x, y });
+}
+
+void EventSenderProxyClientWPE::updateTouchPoint(int index, int x, int y, double)
+{
+    ASSERT(index >= 0 && static_cast<size_t>(index) <= m_touchPoints.size());
```

Comment:
```suggestion
    ASSERT(index >= 0 && static_cast<size_t>(index) < m_touchPoints.size());
```

if `index == m_touchPoints.size()` we'd be accessing out of bounds, wouldn't we?

Reply:
Right, I copied the asserts from the libwpe implementation, that should be fixed in a different patch